cluster-autoscaler/core/scale_down.go

```diff
@@ -64,6 +64,8 @@ const (
 	// PodEvictionHeadroom is the extra time we wait to catch situations when the pod is ignoring SIGTERM and
 	// is killed with SIGKILL after MaxGracefulTerminationTime
 	PodEvictionHeadroom = 30 * time.Second
+	// MaxSoftTaintTime limits single run of SoftTaintUnneededNodes
+	MaxSoftTaintTime = 3 * time.Second
```

Comment:
The comment is fine, but variable name is misleading. I would name it `SoftTaintUnnededNodesRuntimeLimit`